perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -325,9 +348,29 @@ public void updateMailById(PerunSession sess, ApplicationMail mail) throws FormN
 		}
 
 		for (Locale loc : mail.getHtmlMessage().keySet()) {
-			MailText htmlText = mail.getHtmlMessage(loc);
+			MailText htmlMessage = mail.getHtmlMessage(loc);
+			// Check if html text contains invalid tags
+			String inputSubject = htmlMessage.getSubject();
+			String inputText = htmlMessage.getText();
+
+			if (inputSubject != null && inputText != null) {
+				// Check if html text contains invalid tags (subject and text)
+				HTMLParser parser = new HTMLParser()
+					.sanitizeHTML(inputSubject)
+					.checkEscapedHTML();
+				if (!parser.isInputValid()) {
+					throw new InvalidHtmlInputException("HTML Subject contains unsafe HTML tags or styles. Remove them and try again.", parser.getEscaped());
+				}
+				htmlMessage.setSubject(parser.getEscapedHTML());
+
```

Comment:
You need to include the second sanitization call here (which is why the tests are failing). Also, I believe it would be better practice to create a new HTMLParser instance for EACH string.

Reply:
You are right it is a good practice to do new instance per string